gbus/abstractions.go

```diff
@@ -122,6 +122,10 @@ type Saga interface {
 	New() Saga
 }
 
+type SagaIDProvider interface {
```

Comment:
I think a better naming would reflect the fact that this interface is correlating between messages and saga instances.
In addition, let's reduce the scope of the variable that is passed in from gbus.BusMessage to gbus.Message.
Last but not least, I think this should return a function as a return value so we won't store saga instances on the Def or need to create new instances . in order to invoke this functionality.

I suggest something along the lines of the following

type CustomSagaCorrelation interface{
   Correlator() func(message Message) (string error)
}